haproxy/templates/configmap-redisk8s.yaml

```diff
@@ -0,0 +1,69 @@
+{{- if eq .Values.haproxy.config "redisk8s"}}
+{{- $redisk8s := .Values.haproxy.redisk8s -}}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: "{{ include "haproxy.fullname" . }}-redisk8s"
+  labels:
+    app.kubernetes.io/name: {{ include "haproxy.name" . }}
+    helm.sh/chart: {{ include "haproxy.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+data:
+  haproxy.cfg: |-
+    global
+      log stdout local0
+
+    resolvers mydns
+      parse-resolv-conf
+      accepted_payload_size 8192 # allow larger DNS payloads
+
+    defaults
+      log global
+
+      option dontlognull
+      timeout connect 5s
+      timeout client 10s
+      timeout server 10s
+
+    {{- if .Values.metrics.enabled }}
+    frontend stats
+      mode http
+      bind *:9000
+
+      stats enable
+      stats uri /stats
+      stats refresh 3s
+      stats admin if TRUE
+
+      option http-use-htx
+      http-request use-service prometheus-exporter if { path /metrics }
+    {{ end }}
+
+    frontend redis
+      bind *:{{ .Values.haproxy.frontendPort }}
+      mode tcp
+      option clitcpka
+      default_backend redis-nodes
+
+    backend redis-nodes
+      mode tcp
+
+      {{ if $redisk8s.check.enabled }}
+      option tcp-check
+      tcp-check connect
+      tcp-check send AUTH\ {{ $redisk8s.check.auth }}\r\n
```

Comment:
Let's use an environment variable here and mount it from a secret.

Reply:
Also, AUTH should be optional. If no password/existingSecret is provided, just skip these two lines.

Reply:
the auth is now only done if a password or existingSecret is provided